test/19_apps.ts

```diff
@@ -20,23 +23,31 @@ describe('Apps', function () {
 			fx = await fixtures.load('19-apps');
 
 			admin = fx.users.admin;
-			applicationId = fx.applications.app1.id;
-
-			supervisorAppUuid = fx.applications.supervisorApp.uuid;
+			userApp = fx.applications.app1;
+			supervisorApp = fx.applications.supervisorApp;
+			supervisorRelease2 = fx.releases.supervisorRelease2;
 
 			deviceWithSupervisor = await fakeDevice.provisionDevice(
 				admin,
-				applicationId,
+				userApp.id,
 				'balenaOS 2.3.0+rev1',
 				'1.0.1',
-				fx.releases.supervisorRelease1.id,
```

Comment:
Ends up that all that this was not respected and the API was overwriting it w/ the sv release that it found based on `1.0.1`. On top of this `provisionDevice` should probably be changed to be using `register/device` rather than orchestrating OData requests, so that it's more realistic, and in that case providing the target sv release will not be a case anyway.